containers/daap-python-base/src/var/task/data_platform_paths.py

```diff
@@ -77,71 +109,130 @@ def get_account_id() -> str:
     return boto3.client("sts").get_caller_identity()["Account"]
 
 
-class DataProductConfig:
+@dataclass
+class DataProductElement:
     """
-    Configures the name, S3 paths, and athena table names for a data product.
+    A dataset within the data product. The curated data for each element
+    is queryable in its own athena table.
     """
 
-    def __init__(self, name: str, table_name: str, bucket_name: str | None = None):
-        """
-        Generate all the paths based on the data product name and a table name
-        """
-        if bucket_name is None:
-            bucket_name = get_bucket_name()
+    name: str
+    data_product: DataProductConfig
 
-        self.name = name
+    @staticmethod
+    def load(name, data_product_name):
+        data_product = DataProductConfig(name=data_product_name)
+        return DataProductElement(data_product=data_product, name=name)
 
-        self.raw_data_prefix = BucketPath(
-            bucket=bucket_name, key=os.path.join("raw_data", name, table_name) + "/"
+    @property
+    def raw_data_prefix(self):
+        return BucketPath(
+            bucket=self.data_product.raw_data_bucket,
+            key=os.path.join("raw_data", self.data_product.name, self.name) + "/",
         )
 
-        self.curated_data_prefix = BucketPath(
-            bucket_name,
+    @property
+    def curated_data_prefix(self):
+        return BucketPath(
+            self.data_product.curated_data_bucket,
             os.path.join(
                 "curated_data",
-                f"database_name={name}",
-                f"table_name={table_name}",
+                f"database_name={self.data_product.name}",
+                f"table_name={self.name}",
             )
             + "/",
         )
 
-        self.raw_data_table = QueryTable(
-            database=RAW_DATABASE_NAME, name=f"{table_name}_raw"
-        )
-        self.curated_data_table = QueryTable(database=name, name=table_name)
+    def raw_data_table_unique(self):
+        suffix = uuid4()
```

Comment:
its the name of the file

Reply:
@tom-webber It could be, but I was thinking of this as a separate randomiser that is independent of the filename.

This is used to name a temporary table for the lambda to read from when populating or resyncing curated data.

The idea behind adding some random string was just to handle the edge case where the lambda runs twice in succession for the same data product element. That could be two different extractions or it could be the same extraction being repeated twice for some reason.